feat: first pass on pgrx bump

osintalex opened the pull request:
Probably lots wrong here now but just wanted to get the ball started and have some feedback

osintalex commented on src/bigint.rs:
```diff
@@ -44,7 +45,7 @@ mod tests {
     #[pg_test]
     fn type_has_operators() {
-        let type_oid_datum = Spi::get_one_with_args::<u32>(
+        let type_oid_datum = Spi::get_one_with_args::<Oid>(
```
Not sure what changed that meant tests now require this to pass

osintalex commented on src/text.rs:
```diff
@@ -176,13 +176,14 @@ mod tests {
+        assert_eq!(expected_len, actual_len);
     }
 
+    // these currently fail for https://github.com/pgcentralfoundation/pgrx/issues/1860
```
More specifically, it was unable to find the postgres type `enquo_set_ore_32_8`. That github issues suggests this is an error for vectors. I tried adding a search path macro as in that issue but still couldn't get this to pass :/

osintalex replied:
Have addressed this - the type `enquo_set_ore_32_8` was incorrect, should have been `enquo_ore_32_8`. However, the tests are still failing even though the type works. Were these passing before? not too sure how they are supposed to work.

mpalmer replied:
You make an excellent point.  I have no idea what I was thinking, because the current `main` is broken for me, too, and I can't see how it could ever have worked.  It looks like you've fixed up those problems in your PR already, though, which I apologise you had to do in addition to the upgrade work.

That being said, the test suite *does* pass on `main^` for me, which used the `enquo_set_*` types.  I'm pretty sure the type of that variable should be changed to `enquo_kith_ore_32_8`, rather than `enquo_ore_32_8`, since that's what I renamed the `enquo_set_ore_32_8` type to in the most recent commit.  I did document [what the dickens a Kith even is](https://github.com/enquo/enquo-core/blob/main/rust/src/datatype/kith.rs), if you're curious, although I added those docs after the last release of `enquo-core`, so it doesn't show up on docs.rs.  Perhaps I should cut a new release of that...

osintalex replied:
No problem! Hehe, I like the name `Kith`. Thinks this is all fixed in latest commit - all made a bit more sense when I looked at the pg operator macro functions in the `ore.rs` module!

osintalex commented:
Hey! Any thoughts on this :-)

On Sat, Nov 23, 2024 at 00:00, Matt Palmer ***@***.***(mailto:On Sat, Nov 23, 2024 at 00:00, Matt Palmer <<a href=)> wrote:

> @mpalmer commented on this pull request.
>
> ---------------------------------------------------------------
>
> In [src/text.rs](https://github.com/enquo/pg_enquo/pull/5#discussion_r1854777862):
>
>>      }
>
> +    // these currently fail for https://github.com/pgcentralfoundation/pgrx/issues/1860
>
> You make an excellent point. I have no idea what I was thinking, because the current main is broken for me, too, and I can't see how it could ever have worked. It looks like you've fixed up those problems in your PR already, though, which I apologise you had to do in addition to the upgrade work.
>
> That being said, the test suite does pass on main^ for me, which used the enquo_set_* types. I'm pretty sure the type of that variable should be changed to enquo_kith_ore_32_8, rather than enquo_ore_32_8, since that's what I renamed the enquo_set_ore_32_8 type to in the most recent commit. I did document [what the dickens a Kith even is](https://github.com/enquo/enquo-core/blob/main/rust/src/datatype/kith.rs), if you're curious, although I added those docs after the last release of enquo-core, so it doesn't show up on docs.rs. Perhaps I should cut a new release of that...
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/enquo/pg_enquo/pull/5#discussion_r1854777862), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/APHRW4TJTORP2LCPPENDN2L2B6SPBAVCNFSM6AAAAABSCWSTBWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDINJVHAZDAOJQGE).
> You are receiving this because you authored the thread.Message ID: ***@***.***>

mpalmer commented:
Thanks for the bump -- I didn't get a notification that you'd pushed a fixing commit (or I missed it somehow).  It all looks good to me, if you're comfortable with the state of things, I'll merge it.

osintalex commented:
Oooh no worries! I just bumped after thinking how useful this project would be when reading on the oracle breaches.

Looks good to me, happy to merge

On Fri, Apr 4, 2025 at 01:17, Matt Palmer ***@***.***(mailto:On Fri, Apr 4, 2025 at 01:17, Matt Palmer <<a href=)> wrote:

> Thanks for the bump -- I didn't get a notification that you'd pushed a fixing commit (or I missed it somehow). It all looks good to me, if you're comfortable with the state of things, I'll merge it.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/enquo/pg_enquo/pull/5#issuecomment-2777191209), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/APHRW4QFANKCPQZU5UDLTWL2XXFSDAVCNFSM6AAAAABSCWSTBWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZXGE4TCMRQHE).
> You are receiving this because you authored the thread.Message ID: ***@***.***>
>
> [mpalmer] mpalmer left a comment [(enquo/pg_enquo#5)](https://github.com/enquo/pg_enquo/pull/5#issuecomment-2777191209)
>
> Thanks for the bump -- I didn't get a notification that you'd pushed a fixing commit (or I missed it somehow). It all looks good to me, if you're comfortable with the state of things, I'll merge it.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/enquo/pg_enquo/pull/5#issuecomment-2777191209), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/APHRW4QFANKCPQZU5UDLTWL2XXFSDAVCNFSM6AAAAABSCWSTBWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZXGE4TCMRQHE).
> You are receiving this because you authored the thread.Message ID: ***@***.***>
